packages/react-native-gesture-handler/src/v3/hooks/callbacks/js/useGestureStateChangeEvent.ts

```diff
@@ -0,0 +1,15 @@
+import { CallbackHandlers } from '../../../types';
+import { onGestureHandlerStateChange } from '../onGestureHandlerStateChange';
+
+export function gestureStateChangeEvent(handlerTag: number, config: any) {
+  const { onBegin, onStart, onEnd, onFinalize } = config;
+
+  const handlers: CallbackHandlers = {
+    ...(onBegin ? { onBegin } : {}),
+    ...(onStart ? { onStart } : {}),
+    ...(onEnd ? { onEnd } : {}),
+    ...(onFinalize ? { onFinalize } : {}),
+  };
```

Comment:
Also, I will probably extract this into separate function since both, Reanimated and JS use the same logic. Just sayin' 😅

Reply:
Done in [b3ef022](https://github.com/software-mansion/react-native-gesture-handler/pull/3682/commits/b3ef022d7c4edbd1960818794f5b683cfad28e6a)